crates/ruma-appservice-api/src/event/push_events.rs

```diff
@@ -60,6 +74,32 @@ pub mod v1 {
             Self {}
         }
     }
+    /// Information on E2E device updates.
+    #[derive(Clone, Debug, Default, Deserialize, Serialize)]
+    #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+    pub struct DeviceLists {
```

Comment:
This definition as well as the impl block should also be behind `#[cfg(feature = "unstable-msc3202")]`.